rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -152,6 +154,23 @@ impl HyperlaneRocksDB {
         Ok(true)
     }
 
+    /// Store the merkle tree insertion event, and also store a mapping from message_id to leaf_index
+    pub fn process_tree_insertion(&self, insertion: &MerkleTreeInsertion) -> DbResult<bool> {
+        if let Ok(Some(_)) = self.retrieve_merkle_leaf_index_by_message_id(&insertion.message_id())
```

Comment:
isn't this not allowing double insertions?

Reply:
great point, yes